app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java

```diff
@@ -9,43 +9,58 @@
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_SERVER;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_TARGET_DIR;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_USERNAME;
+import static org.bostwickenator.ftpuploader.SettingsStore.settingsStore;
+
 class FtpClient {
 
     private final FTPClient ftpClient = new FTPClient();
     private final static String EMPTY_STRING = "";
-    SettingsStore settingsStore = SettingsStore.getSettingsStore();
+    private SettingsStore settingsStore = settingsStore();
 
     public void connect() {
 
         try {
-            ftpClient.connect(getStringWithEmptyAsDefault(SettingsActivity.SETTING_SERVER), Integer.parseInt(getStringWithEmptyAsDefault(SettingsActivity.SETTING_PORT)));
-            ftpClient.login(getStringWithEmptyAsDefault(SettingsActivity.SETTING_USERNAME), getStringWithEmptyAsDefault(SettingsActivity.SETTING_PASSWORD));
-            if (settingsStore.getBoolean(SettingsActivity.SETTING_PASSIVE, false)) {
-                ftpClient.enterLocalPassiveMode();
+            this.ftpClient.connect(getStringWithEmptyAsDefault(SETTING_SERVER), parseInt(this.settingsStore.getString(SETTING_PORT, "21")));
+            this.ftpClient.login(getStringWithEmptyAsDefault(SETTING_USERNAME), getStringWithEmptyAsDefault(SETTING_PASSWORD));
+            if (this.settingsStore.getBoolean(SETTING_PASSIVE, false)) {
+                this.ftpClient.enterLocalPassiveMode();
+            }
+            this.ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+
+            if (!getStringWithEmptyAsDefault(SETTING_TARGET_DIR).equals(EMPTY_STRING)) {
+                if (!this.ftpClient.changeWorkingDirectory(getStringWithEmptyAsDefault(SETTING_TARGET_DIR))) {
+                    Logger.error("failed changing to target dir");
+                }
             }
-            ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
         } catch (Exception e) {
             Logger.error(e.toString());
         }
     }
     
     private String getStringWithEmptyAsDefault(String key) {
-        return settingsStore.getString(key, EMPTY_STRING);
+        return this.settingsStore.getString(key, EMPTY_STRING);
     }
 
     public void createAndUseAlbumForSubsequentOperations(String name) throws IOException {
-        ftpClient.makeDirectory(name);
-        ftpClient.changeWorkingDirectory(name);
+        this.ftpClient.makeDirectory(name);
+        this.ftpClient.changeWorkingDirectory(name);
     }
 
     public boolean storeFile(File file) throws IOException {
         FileInputStream fis = new FileInputStream(file);
-        boolean result = ftpClient.storeFile(file.getName(), fis);
+        boolean result = this.ftpClient.storeFile(file.getName(), fis);
         fis.close();
         return result;
     }
 
     public void disconnect() throws IOException {
-        ftpClient.disconnect();
+        this.ftpClient.disconnect();
```

Comment:
Appending `this` isn't strictly necessary here, it should really only be used in cases where you want to assign a property from a constructor parameter.

Shame automatic assignment of constructor parameters to the relevant property isn't a thing anyway in java..  Maybe Java should learn from Typescript? ;)  [/unrelated rant]

Reply:
Also matter of preference. Better than to use i.e. mFtpClient as somewhere else in the code base. Ilike it because I can see on one shot that this is an instance var and not something else defined in the scope of the method.